site/profile/manifests/core/velero.pp

```diff
@@ -0,0 +1,20 @@
+# @summary
+#   Velero cli instalation and autocompletion
+
+class profile::core::velero {
+  $runif  = 'test ! -f /etc/bash_completion.d/velero'
+  $command = 'velero completion bash > /etc/bash_completion.d/velero'
+
+  vcsrepo { '/usr/share/velero':
+    ensure   => present,
+    provider => git,
+    source   => 'git://github.com/vmware-tanzu/velero.git',
+    revision => 'v1.4.2',
+  }
+  exec { $command:
```

Comment:
The problem with looking for the existence of a file means that the content will never be updated, even if the version of the application changes.  Putting `echo 'source <(velero completion bash)' ` into a profile.d file eliminates the `exec` resource and should has better forward compatibility.